libbeat/template/template.go

```diff
@@ -193,26 +196,23 @@ func (t *Template) GetPattern() string {
 // The default values are taken from the default variable.
 func (t *Template) Generate(properties common.MapStr, dynamicTemplates []common.MapStr) common.MapStr {
 	keyPattern, patterns := buildPatternSettings(t.esVersion, t.GetPattern())
-
-	return common.MapStr{
+	template := common.MapStr{
 		keyPattern: patterns,
-
-		"mappings": buildMappings(
-			t.beatVersion, t.esVersion, t.beatName,
-			properties,
-			append(dynamicTemplates, buildDynTmpl(t.esVersion)),
-			common.MapStr(t.config.Settings.Source),
-		),
-
-		"order": 1,
-
+		"order":    t.order,
 		"settings": common.MapStr{
 			"index": buildIdxSettings(
 				t.esVersion,
 				t.config.Settings.Index,
 			),
 		},
 	}
+
+	template["mappings"] = buildMappings(
+		t.beatVersion, t.esVersion, t.beatName,
+		properties,
+		append(dynamicTemplates, buildDynTmpl(t.esVersion)),
+		common.MapStr(t.config.Settings.Source))
```

Comment:
now that we always add 'mappings' again, let's move it back to the construction of the map.